libs/shared/contentful/src/lib/contentful.stripe-mapper.ts

```diff
@@ -0,0 +1,234 @@
+          this.stripeMetadata[StripeMetadataKeys.NewsletterSlug]
+            .split(',')
+            .sort()
+            .join(',')
+        );
+      default:
+        return this.stripeMetadata[stripeMetadataKey];
+    }
+  }
+
+  /**
+   * Return the Contentful config value for the provided Stripe metadata key,
+   * in the format expected by Stripe metadata
+   */
+  getContentfulForMetadataKey(stripeMetadataKey: StripeMetadataKeys) {
+    switch (stripeMetadataKey) {
+      case StripeMetadataKeys.DetailsLine1:
+        return this.purchaseDetails.details[0];
+      case StripeMetadataKeys.DetailsLine2:
+        return this.purchaseDetails.details[1];
+      case StripeMetadataKeys.DetailsLine3:
+        return this.purchaseDetails.details[2];
+      case StripeMetadataKeys.DetailsLine4:
+        return this.purchaseDetails.details[3];
+      case StripeMetadataKeys.Subtitle:
+        return this.purchaseDetails.subtitle;
+      case StripeMetadataKeys.WebIcon:
+        return this.purchaseDetails.webIcon;
+      case StripeMetadataKeys.PrivacyNotice:
+        return this.commonContent.privacyNoticeUrl;
+      case StripeMetadataKeys.PrivacyNoticeDownload:
+        return this.commonContent.privacyNoticeDownloadUrl;
+      case StripeMetadataKeys.ToS:
+        return this.commonContent.termsOfServiceUrl;
+      case StripeMetadataKeys.ToSDownload:
+        return this.commonContent.termsOfServiceDownloadUrl;
+      case StripeMetadataKeys.CancellationURL:
+        return this.commonContent.cancellationUrl;
+      case StripeMetadataKeys.EmailIcon:
+        return this.commonContent.emailIcon;
+      case StripeMetadataKeys.SuccessAction:
+        return this.commonContent.successActionButtonUrl;
+      case StripeMetadataKeys.SuccessActionLabel:
+        return this.commonContent.successActionButtonLabel;
+      case StripeMetadataKeys.NewsletterLabel:
+        return this.commonContent.newsletterLabelTextCode;
+      case StripeMetadataKeys.NewsletterSlug: {
+        return (
+          this.commonContent.newsletterSlug &&
+          [...this.commonContent.newsletterSlug].sort().join(',')
+        );
+      }
+      default:
+        return undefined;
+    }
+  }
+
+  /**
+   * Merges Stripe and Contentful values where appropriate, and logs fields where
+   * Contentful does not match Stripe metadata values
+   */
+  mergeStripeAndContentful() {
+    const mappedMetadata: { [key in StripeMetadataKeys]?: string } = {};
+    Object.values(StripeMetadataKeys).forEach((key) => {
+      mappedMetadata[key] = this.mapField(
+        key,
+        this.getStripeForMetadataKey(key),
+        this.getContentfulForMetadataKey(key)
+      );
+    });
+
+    return {
+      metadata: mappedMetadata,
+      errorFields: this.errorFields,
+    };
+  }
+}
+
+/**
+ * Class to fetch Contentful config and map Contentful config to
+ * Stripe metadata for an array of Stripe Plans.
+ */
+export class ContentfulStripeMapper {
```

Comment:
Feels like this should be named with `Manager` or `Service` at the end - I believe this falls under service given it's responsibilities?